### ISSUE TITLE
Use `<form>`s for form experiences

### ISSUE THREAD

**Issue by emyl3** [CDCgov/prime-simplereport#5060]
Throughout the app, we have forms but never use the semantic element for forms: `<form>`.

This occurs in:
- [ ] `QuestionsForm.tsx`
- [ ] `PersonalDetailsForm.tsx`
- [ ] `PersonForm.tsx`
- [ ] `SelfRegistrationForm.tsx`

**Action requested**
Use `<form>` for forms.

**Comment by emyl3** [comment]
bit of background info on this: https://skylight-hq.slack.com/archives/C03KQ79EB16/p1674150965735509?thread_ts=1674144631.536469&cid=C03KQ79EB16

**Comment by emmastephenson** [comment]
We are using the USWDS form components - need to see if these resolve to form groups under the hood. Should also double check the Deque form guidance.
